flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/validation/DefaultDeploymentValidator.java

```diff
@@ -245,6 +241,19 @@
 
         JobSpec oldJob = oldSpec.getJob();
         JobSpec newJob = newSpec.getJob();
+
+        if (lastReconciledSpec(deployment) != null
+                && ReconciliationUtils.triggerRestart(deployment)) {
+            if (newJob.getState() != JobState.RUNNING) {
+                return Optional.of(
+                        String.format("Cannot restart job to %s state", newJob.getState()));
+            }
+            if (oldJob.getState() != JobState.RUNNING) {
+                return Optional.of(
+                        String.format("Cannot restart job in %s state", oldJob.getState()));
+            }
+        }
```

Comment:
I think we can remove these checks together with the `ReconciliationUtils.triggerRestart(deployment))` I dont think we are achieving anything and we actually restrict cases when the user want to restart a failing job

Reply:
This validation is try to guard that job should target `RUNNING` state and also in `RUNNING` state when user try to use the restartNonce. The `RUNNING` state is the desire state of the job not the observed state here. Otherwise, what will we do if the job is in `SUSPENDED` state and change the `restartNonce` ?

Reply:
nothing will happen, but why would anything happen anyways :) without the validaiton the restartNonce will behave like any other spec change

Reply:
Instead of adding this specific validation for the `restartNonce` field we could discuss whether we should allow spec changes in suspended state. I think it is fine

Reply:
> but why would anything happen anyways

I think the `restartNonce` have the semantic of bring the job to a `start/running` status somehow? If not go with this check it will work like ask job to restart once, but it still suspend :).

> we could discuss whether we should allow spec changes in suspended state

If JobState changes from suspended to suspended, with other spec changes, It will not take effect, I think it will not bring bad impact to allow the change. The problem of the `restartNonce` field change in suspended state is only the problem of the semantic, as I said first.

Reply:
I think the way you implemented it the restartNonce will not change the desired state. And I think it's good like this. This is just a field to trigger restart without having to change the actual spec.

Reply:
About the removal of the validation, I want to keep it to Monday to wait for more inputs. Not insist on it, Just want to hear more opinions on it :)

Reply:
All I am trying to say, is that the way it is implemented, changing the `restartNonce` field works as any other spec change and it triggers a simple upgrade of the deplyoment. (this is perfectly good)

Given this, I don't see the point in introducing restrictions on this field as we also dont have restrictions on when an upgrade can be triggered. I just want to keep it conistent and simple.

We can of course wait for others to chime in here as well:
@wangyang0918 @morhidi @tweise

Reply:
I lean to make `restartNonce` working as a normal fields and do not have special validation.

Reply:
@gyfora @wangyang0918 Thanks for your suggestion, I have removed the validation accordingly